File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/c/HexagonalNum.c
#include "java_arrays.h"
#include <stdint.h>

/*@
  assigns \nothing;
  ensures \result == (int32_t)((integer)left - right);
*/
static int32_t java_sub(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left - (uint32_t)right);
}

/*@
  assigns \nothing;
  ensures \result == (int32_t)((integer)left * right);
*/
static int32_t java_mul(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left * (uint32_t)right);
}

/*@
  assigns \nothing;
  ensures \result == (int32_t)((integer)n * (2 * n - 1));
*/
int32_t hexagonalNum(int32_t n) {
    return java_mul(n, java_sub(java_mul(2, n), 1));
}

File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/java/HexagonalNum.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;


class HexagonalNum {
    
    /*@ public normal_behavior
      @ assignable \nothing;
      @ ensures \result == \java_math(n * (2 * n - 1));
      @*/
    //@ code_java_math
    public static int hexagonalNum(int n) {
        int ans = n * (2*n - 1);
        return ans;
    }
}

